optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/common/AbstractRestServiceImpl.java

```diff
@@ -12,9 +12,98 @@
 
     protected void validateTenantIdParameter(Integer tenantId, AbstractPersistable persistable) {
         if (!Objects.equals(persistable.getTenantId(), tenantId)) {
-            throw new IllegalStateException("The tenantId (" + tenantId
-                    + ") does not match the persistable (" + persistable + ")'s tenantId (" + persistable.getTenantId() + ").");
+            throw new IllegalStateException("The tenantId (" + tenantId + ") does not match the persistable (" + persistable + ")'s tenantId (" + persistable.getTenantId() + ").");
         }
     }
 
+    protected List<Long> getTableCountAndVersionSum(Integer tenantId, EntityManager entityManager, Class<? extends AbstractPersistable> entityClass) {
```

Comment:
Instead of the hash approach, we'll look into the optimistic version number approach in our next meeting.